eudemo/eudemo/maintenance/lower_port/designer.py

```diff
@@ -0,0 +1,588 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+EU-DEMO Lower Port
+"""
+from typing import Dict, Iterable, List, Tuple, Type, Union
+
+import numpy as np
+
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import ParameterFrame
+from bluemira.geometry.base import BluemiraGeo
+from bluemira.geometry.constants import D_TOLERANCE
+from bluemira.geometry.error import GeometryError
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.tools import (
+    boolean_cut,
+    boolean_fuse,
+    distance_to,
+    make_polygon,
+    offset_wire,
+)
+from bluemira.geometry.wire import BluemiraWire
+from eudemo.maintenance.lower_port.parameterisations import LowerPortDesignerParams
+
+
+class LowerPortDesigner(Designer):
```

Comment:
Not entirely clear to me what the scope of this designer is; it seems there are a lot of return values.

For the upper port, we have designer that just makes the keep-out zone, and then the builder handles everything relating to the y dimension. This is because we always make the ports as wide as they can be in y. Perhaps a similar separation can be achieved here.

This might be the best way to do it, but I think we can get away with a simpler parameterisation for the lower port keep out zone. Something like this:
![image](https://user-images.githubusercontent.com/26097289/234484667-ad6619b4-4230-43a0-99f0-3b6101ed4a18.png)
To which you'd have to add various offsets for thicknesses and gaps of and between the thermal shield and VV.